web_crawler/web_crawler/middlewares.py

```
@@ -0,0 +1,103 @@
+# Define here the models for your spider middleware
+#
+# See documentation in:
+# https://docs.scrapy.org/en/latest/topics/spider-middleware.html
+
+from scrapy import signals
+
+# useful for handling different item types with a single interface
+from itemadapter import is_item, ItemAdapter
```

Ruff (0.8.0)

`itemadapter.is_item` imported but unused

Remove unused import

(F401)